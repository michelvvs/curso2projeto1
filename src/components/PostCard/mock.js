export const postCardPropsMock = {
  title: 'title1',
  body: 'body1',
  id: 2,
  cover: 'img/img.png',
  comments: [],
};
