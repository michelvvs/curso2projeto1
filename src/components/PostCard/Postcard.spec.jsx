import { render, screen } from '@testing-library/react';
import { PostCard } from '.';
import { postCardPropsMock } from './mock';

const props = postCardPropsMock;

describe('<PostCard />', () => {
  it('should render Postcard correctly', () => {
    render(<PostCard {...props} />);

    expect(screen.getByRole('img', { name: /title/i })).toHaveAttribute('src', props.cover);
    expect(screen.getByRole('heading', { name: /title/i })).toBeInTheDocument();
    expect(screen.getByText('body1')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not render posts', () => {
    render(<PostCard />);
    expect(screen.queryByRole('heading', { name: /title/i })).not.toBeInTheDocument();
  });
});
