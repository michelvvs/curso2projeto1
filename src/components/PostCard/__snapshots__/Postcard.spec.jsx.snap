// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <div
    class="post-content"
  >
    <img
      alt="title1"
      src="img/img.png"
    />
    <h1>
      title1
       (
      2
      )
    </h1>
    <p>
      body1
    </p>
    <div
      class="comments"
    >
      <p>
        <strong>
          coment√°rios: 
        </strong>
        0
         
      </p>
      <p>
        eita
      </p>
    </div>
  </div>
</div>
`;
